spring-integration-http/src/main/java/org/springframework/integration/http/config/IntegrationGraphControllerRegistrar.java

```diff
@@ -56,6 +61,16 @@ public void registerBeanDefinitions(AnnotationMetadata importingClassMetadata, B
 					new RootBeanDefinition(IntegrationGraphServer.class));
 		}
 
+		String[] allowedOrigins = (String[]) annotationAttributes.get("allowedOrigins");
+		if (allowedOrigins != null && allowedOrigins.length > 0) {
+			AbstractBeanDefinition controllerCorsConfigurer =
+				BeanDefinitionBuilder.genericBeanDefinition(IntegrationGraphCorsConfigurer.class)
+					.addConstructorArgValue(annotationAttributes.get("value"))
+					.addConstructorArgValue(allowedOrigins)
+					.getBeanDefinition();
+			BeanDefinitionReaderUtils.registerWithGeneratedName(controllerCorsConfigurer, registry);
```

Comment:
I wonder why I didn't come up with this solution before....
See https://jira.spring.io/browse/INT-4035
To be honest I don't remember what obstacle I had bumped that time that ended up with just documentation...

I won't mind to merge and cherry-pick but would like to see some documentation reflection as well.
Thanks.